src/SOS/SOS.InstallHelper/InstallHelper.cs

```diff
@@ -0,0 +1,307 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Security;
+
+namespace SOS
+{
+    /// <summary>
+    /// Functions to install and configure SOS from the package containing this code.
+    /// </summary>
+    public sealed class InstallHelper
+    {
+        /// <summary>
+        /// Well known location to install SOS. Defaults to $HOME/.dotnet/sos on xplat and %USERPROFILE%/.dotnet/sos on Windows.
+        /// </summary>
+        public string InstallLocation { get; set; }
+
+        /// <summary>
+        /// On Linux/MacOS, the location of the lldb ".lldbinit" file. Defaults to $HOME/.lldbinit.
+        /// </summary>
+        public string LLDBInitFile { get; set; }
+
+        /// <summary>
+        /// If true, enable the symbol server support when configuring lldb.
+        /// </summary>
+        public bool EnableSymbolServer { get; set; } = true;
+
+        /// <summary>
+        /// The source path from which SOS is installed. Default is OS/architecture (RID) named directory in the same directory as this assembly.
+        /// </summary>
+        public string SOSSourcePath { get; set; }
+
+        /// <summary>
+        /// Console output delegate
+        /// </summary>
+        private Action<string> m_writeLine;
+
+        /// <summary>
+        /// Create an instance of the installer.
+        /// </summary>
+        /// <exception cref="PlatformNotSupportedException">unknown operating system</exception>
+        /// <exception cref="InvalidOperationException">environment variable not found</exception>
+        public InstallHelper(Action<string> writeLine)
+        {
+            m_writeLine = writeLine;
+            string home = null;
+            string os = null;
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows)) 
+            {
+                home = Environment.GetEnvironmentVariable("USERPROFILE");
+                if (string.IsNullOrEmpty(home)) {
```

Comment:
I'd recommend logging the env var and testing to see if it is a valid path. APIs would fail with exceptions later, but the error handling won't be nearly as clear.

Reply:
I don't do any more validation in the constructor since the paths being initialized can be changed later by the command line parsing. There might be cases that the SOSSourcePath (like during testing) might not be valid until Install()/Uninstalled are called. 

I think there is enough validation in Install()/Uninstall() with good enough error handling.